### ISSUE TITLE
for (i of []) fails for variable declared in outer scope

### ISSUE THREAD

**Issue by jdom** [trullock/NUglify#253]
When minifying the following code:

```
let i;
for (i of ["foo"]) {
   console.log(i);
};
```

It outputs:
```
let i;for(i["foo"];;)console.log(i)
```

But the expected is:
```
let i;for(i of["foo"])console.log(i)
```

Note that for..of works correctly when defining and initializing the variable in the same line (`for (const i of [])` ) whether I use `let` or `const`. It also works correctly in the `for (i in [])` case, it's just with the for..of case that it fails.

**Comment by trullock** [comment]
Ok great thanks for this. Should be an easy fix if it already works with in

**Comment by jdom** [comment]
Yeah, it shouldn't be too hard given the support for "in". I did take a short look when I created the associated repro PR,. It wasn't as straightforward as a simple replacement in 1 place, and given I'm not familiar with it, I didn't jump on it yet, and decided to fix the usage of it in our codebase for the time being (TBH, I don't know why we were defining the variable outside of the `for` in our case). Ours is a massive codebase, and we had just 1 occurrence of this, which resulted in an infinite loop at runtime when changing the TypeScript target from es5 to es6+ (due to this small bug in the minifier). We are still in the process of making our codebase work correctly with es6, so I might have a few more bugs like this (hopefully no though)
Thanks for this project! We were using AjaxMin before when targeting es5, so this was a simple drop in replacement at this time so we could focus on other bigger things :)

**Comment by trullock** [comment]
Fixed in 1.13.11

**Comment by jdom** [comment]
> I take beer donations :)

Gladly! how do I donate? I couldn't find any donation link

**Comment by trullock** [comment]
> > I take beer donations :)
> 
> Gladly! how do I donate? I couldn't find any donation link

https://paypal.me/trullock :)

**Comment by jdom** [comment]
Check out your PayPal account. Cheers! 🍺

**Comment by trullock** [comment]
Graciously received, thanks!
